app/api/routes/image_queries.py

```diff
@@ -251,6 +255,21 @@ async def get_image_query(
     return safe_call_api(gl.get_image_query, id=id)
 
 
+def _get_detector_config_by_id(detector_configs: list[DetectorConfig], detector_id: str) -> DetectorConfig | None:
```

Comment:
Could we just make detector_configs a set?

Reply:
I think this basically just looks like collapsing the list into a dict in `load_edge_config()`, but it:
a) removes the need to iterate over all detector configs on the critical path
b) is cleaner, since this method is basically just `my_dict.get(detector_id)`

Reply:
That makes sense to me, definitely makes it easier to work with.